### PR TITLE
Add option for client to verify negotiated key exchange algorithms.

### PR DESCRIPTION
**Pull request by dkocher**

### CODE REVIEW

File: src/main/java/net/schmizz/sshj/sftp/SFTPEngine.java [hunk 1]
```diff
@@ -59,7 +59,12 @@ public SFTPEngine(SessionFactory ssh)
 
     public SFTPEngine(SessionFactory ssh, String pathSep)
             throws SSHException {
-        sub = ssh.startSession().startSubsystem("sftp");
+        this(ssh, pathSep, "sftp");
+    }
+
+    public SFTPEngine(SessionFactory ssh, String pathSep, String subsystemName)
```
Comment (hierynomus) [inline src/main/java/net/schmizz/sshj/sftp/SFTPEngine.java]: Part of this PR? Seems unrelated at first sight.

Comment (dkocher) [inline src/main/java/net/schmizz/sshj/sftp/SFTPEngine.java]: Yes, completly unrelated. I should work on branches that can be pulled.

**Comment by hierynomus** [comment]
Definitely like the idea! If you can remove the unrelated commit ;)

File: src/main/java/net/schmizz/sshj/sftp/SFTPEngine.java (hunk 1 above)
Comment (hierynomus) [inline src/main/java/net/schmizz/sshj/sftp/SFTPEngine.java]: Let me know when you've fixed that :)
